src/ser.rs

```diff
@@ -766,7 +766,11 @@ macro_rules! serialize_float {
         } else {
             write!($this.dst, "{}", $v).map_err(ser::Error::custom)?;
         }
-        if $v % 1.0 == 0.0 {
+        let mut a = $v;
+        while a > 1.0 {
+            a = a - 1.0;
+        }
+        if a == 0.0 {
```

Comment:
To find the remainder, just subtract the divisor from the dividend until you can't anymore, then what is left is the remainder.